### ISSUE TITLE
0.3.16 crash (core dump) but 0.3.15 not

### ISSUE THREAD

**Issue by Krozark** [alphacep/vosk-api#355]
I just updated my dependencies, and using the following code, I get a core dump:

```
from vosk import Model
model_path = ... # the one from here :https://alphacephei.com/vosk/models/vosk-model-fr-0.6-linto.zip
model = Model(model_path)
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:194) Decoding params beam=13 max-active=7000 lattice-beam=6
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:197) Silence phones 1:2:3:4:5:6:7:8:9:10
Illegal instruction (core dumped)
```

I'm using python 3.7.5 on linux mint.

Notice that reverting the version to 0.3.15 make the Madel class instantiation happy (no core dump)

**Comment by nshmyrev** [comment]
What hardware and os do you have

You can identify illegal instruction with gdb:

```
gdb --args python3 ./test.py test.wav
(gdb) run
(gdb) disassemble

```

**Comment by Krozark** [comment]
OS: Linux Mint 19 Cinnamon
Version of Cinnamon : 3.8.9
Kernel : 4.15.0.20-generic
CPU : Intel© Core™ i7-3770K CPU @ 3.50GHz × 4
RAM : 23.5 Go
GPU : NVIDIA Corporation GK106 [GeForce GTX 660]

I run the test using python/example/test_empty.py
model is https://alphacephei.com/vosk/models/vosk-model-fr-0.6-linto.zip

```
gdb --args $(which python) ./test_empty.py 
GNU gdb (Ubuntu 8.1.1-0ubuntu1) 8.1.1
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /media/HDD1/DEV/virtenvs/ava/bin/python...(no debugging symbols found)...done.
(gdb) run
Starting program: /media/HDD1/DEV/virtenvs/ava/bin/python ./test_empty.py
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:194) Decoding params beam=13 max-active=7000 lattice-beam=6
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:197) Silence phones 1:2:3:4:5:6:7:8:9:10

Program received signal SIGILL, Illegal instruction.
0x00007ffff5168a49 in dgbmv_t ()
   from /media/HDD1/DEV/virtenvs/ava/lib/python3.7/site-packages/vosk/_vosk.cpython-37m-x86_64-linux-gnu.so
(gdb) disassemble
Dump of assembler code for function dgbmv_t:
   0x00007ffff51688c0 <+0>:	push   %r15
   0x00007ffff51688c2 <+2>:	push   %r14
   0x00007ffff51688c4 <+4>:	push   %r13
   0x00007ffff51688c6 <+6>:	mov    %r8,%r13
   0x00007ffff51688c9 <+9>:	push   %r12
   0x00007ffff51688cb <+11>:	push   %rbp
   0x00007ffff51688cc <+12>:	mov    %rcx,%rbp
   0x00007ffff51688cf <+15>:	push   %rbx
   0x00007ffff51688d0 <+16>:	mov    %rdx,%rbx
   0x00007ffff51688d3 <+19>:	sub    $0x58,%rsp
   0x00007ffff51688d7 <+23>:	mov    0xa0(%rsp),%r11
   0x00007ffff51688df <+31>:	mov    0xa8(%rsp),%rdx
   0x00007ffff51688e7 <+39>:	mov    %r9,0x18(%rsp)
   0x00007ffff51688ec <+44>:	mov    %rdi,0x8(%rsp)
   0x00007ffff51688f1 <+49>:	mov    0x90(%rsp),%r12
   0x00007ffff51688f9 <+57>:	mov    %rsi,0x30(%rsp)
   0x00007ffff51688fe <+62>:	mov    0x98(%rsp),%r9
   0x00007ffff5168906 <+70>:	vmovsd %xmm0,0x10(%rsp)
   0x00007ffff516890c <+76>:	mov    0xb0(%rsp),%r14
   0x00007ffff5168914 <+84>:	mov    %r11,0x40(%rsp)
   0x00007ffff5168919 <+89>:	mov    %rdx,0x38(%rsp)
   0x00007ffff516891e <+94>:	mov    %r11,0x28(%rsp)
   0x00007ffff5168923 <+99>:	cmp    $0x1,%rdx
   0x00007ffff5168927 <+103>:	je     0x7ffff516896f <dgbmv_t+175>
   0x00007ffff5168929 <+105>:	lea    0xfff(%r14,%rsi,8),%rax
   0x00007ffff5168931 <+113>:	mov    %rsi,%rdi
   0x00007ffff5168934 <+116>:	mov    %r9,0x48(%rsp)
   0x00007ffff5168939 <+121>:	mov    %r14,%rcx
   0x00007ffff516893c <+124>:	mov    0x138fc25(%rip),%r15        # 0x7ffff64f8568
   0x00007ffff5168943 <+131>:	and    $0xfffffffffffff000,%rax
   0x00007ffff5168949 <+137>:	mov    $0x1,%r8d
   0x00007ffff516894f <+143>:	mov    %r11,%rsi
   0x00007ffff5168952 <+146>:	mov    %rax,0x20(%rsp)
   0x00007ffff5168957 <+151>:	mov    (%r15),%rax
   0x00007ffff516895a <+154>:	callq  *0x300(%rax)
   0x00007ffff5168960 <+160>:	mov    %r14,0x28(%rsp)
   0x00007ffff5168965 <+165>:	mov    0x48(%rsp),%r9
   0x00007ffff516896a <+170>:	mov    0x20(%rsp),%r14
   0x00007ffff516896f <+175>:	cmp    $0x1,%r9
   0x00007ffff5168973 <+179>:	je     0x7ffff516899c <dgbmv_t+220>
   0x00007ffff5168975 <+181>:	mov    0x138fbec(%rip),%r15        # 0x7ffff64f8568
   0x00007ffff516897c <+188>:	mov    %r12,%rsi
   0x00007ffff516897f <+191>:	mov    %r14,%rcx
   0x00007ffff5168982 <+194>:	mov    %r9,%rdx
   0x00007ffff5168985 <+197>:	mov    $0x1,%r8d
   0x00007ffff516898b <+203>:	mov    0x8(%rsp),%rdi
   0x00007ffff5168990 <+208>:	mov    %r14,%r12
   0x00007ffff5168993 <+211>:	mov    (%r15),%rax
   0x00007ffff5168996 <+214>:	callq  *0x300(%rax)
   0x00007ffff516899c <+220>:	mov    0x8(%rsp),%rax
   0x00007ffff51689a1 <+225>:	lea    (%rax,%rbx,1),%rdi
   0x00007ffff51689a5 <+229>:	mov    0x30(%rsp),%rax
   0x00007ffff51689aa <+234>:	cmp    %rax,%rdi
   0x00007ffff51689ad <+237>:	mov    %rax,%rdx
   0x00007ffff51689b0 <+240>:	cmovle %rdi,%rdx
   0x00007ffff51689b4 <+244>:	test   %rdx,%rdx
   0x00007ffff51689b7 <+247>:	jle    0x7ffff5168a63 <dgbmv_t+419>
   0x00007ffff51689bd <+253>:	mov    0x18(%rsp),%rax
   0x00007ffff51689c2 <+258>:	mov    0x28(%rsp),%r14
   0x00007ffff51689c7 <+263>:	lea    0x1(%rbx,%rbp,1),%rbp
   0x00007ffff51689cc <+268>:	mov    0x138fb95(%rip),%r15        # 0x7ffff64f8568
   0x00007ffff51689d3 <+275>:	shl    $0x3,%rax
   0x00007ffff51689d7 <+279>:	mov    %rax,0x18(%rsp)
   0x00007ffff51689dc <+284>:	mov    %rbx,%rax
   0x00007ffff51689df <+287>:	sub    %rdx,%rax
   0x00007ffff51689e2 <+290>:	mov    %rax,0x20(%rsp)
   0x00007ffff51689e7 <+295>:	mov    %r12,%rax
   0x00007ffff51689ea <+298>:	mov    %r14,%r12
   0x00007ffff51689ed <+301>:	mov    %rax,%r14
   0x00007ffff51689f0 <+304>:	jmp    0x7ffff5168a01 <dgbmv_t+321>
   0x00007ffff51689f2 <+306>:	nopw   0x0(%rax,%rax,1)
   0x00007ffff51689f8 <+312>:	mov    0x8(%rsp),%rax
   0x00007ffff51689fd <+317>:	lea    (%rax,%rbx,1),%rdi
   0x00007ffff5168a01 <+321>:	test   %rbx,%rbx
   0x00007ffff5168a04 <+324>:	mov    $0x0,%edx
   0x00007ffff5168a09 <+329>:	mov    (%r15),%r10
   0x00007ffff5168a0c <+332>:	mov    $0x1,%r8d
   0x00007ffff5168a12 <+338>:	cmovns %rbx,%rdx
   0x00007ffff5168a16 <+342>:	mov    %rdx,%rcx
   0x00007ffff5168a19 <+345>:	lea    0x0(%r13,%rdx,8),%rsi
   0x00007ffff5168a1e <+350>:	sub    %rbx,%rcx
   0x00007ffff5168a21 <+353>:	cmp    %rbp,%rdi
   0x00007ffff5168a24 <+356>:	cmovg  %rbp,%rdi
   0x00007ffff5168a28 <+360>:	lea    (%r14,%rcx,8),%rcx
   0x00007ffff5168a2c <+364>:	sub    $0x1,%rbx
   0x00007ffff5168a30 <+368>:	add    $0x8,%r12
   0x00007ffff5168a34 <+372>:	sub    %rdx,%rdi
   0x00007ffff5168a37 <+375>:	mov    $0x1,%edx
   0x00007ffff5168a3c <+380>:	callq  *0x308(%r10)
   0x00007ffff5168a43 <+387>:	vmovsd 0x10(%rsp),%xmm1
=> 0x00007ffff5168a49 <+393>:	vfmadd213sd -0x8(%r12),%xmm1,%xmm0
   0x00007ffff5168a50 <+400>:	add    0x18(%rsp),%r13
   0x00007ffff5168a55 <+405>:	vmovsd %xmm0,-0x8(%r12)
   0x00007ffff5168a5c <+412>:	cmp    %rbx,0x20(%rsp)
   0x00007ffff5168a61 <+417>:	jne    0x7ffff51689f8 <dgbmv_t+312>
   0x00007ffff5168a63 <+419>:	cmpq   $0x1,0x38(%rsp)
   0x00007ffff5168a69 <+425>:	je     0x7ffff5168aa8 <dgbmv_t+488>
   0x00007ffff5168a6b <+427>:	mov    0x38(%rsp),%r8
   0x00007ffff5168a70 <+432>:	mov    0x40(%rsp),%rcx
   0x00007ffff5168a75 <+437>:	mov    $0x1,%edx
   0x00007ffff5168a7a <+442>:	mov    0x138fae7(%rip),%rax        # 0x7ffff64f8568
   0x00007ffff5168a81 <+449>:	mov    0x28(%rsp),%rsi
   0x00007ffff5168a86 <+454>:	mov    0x30(%rsp),%rdi
   0x00007ffff5168a8b <+459>:	mov    (%rax),%rax
   0x00007ffff5168a8e <+462>:	mov    0x300(%rax),%rax
   0x00007ffff5168a95 <+469>:	add    $0x58,%rsp
   0x00007ffff5168a99 <+473>:	pop    %rbx
   0x00007ffff5168a9a <+474>:	pop    %rbp
   0x00007ffff5168a9b <+475>:	pop    %r12
   0x00007ffff5168a9d <+477>:	pop    %r13
   0x00007ffff5168a9f <+479>:	pop    %r14
   0x00007ffff5168aa1 <+481>:	pop    %r15
   0x00007ffff5168aa3 <+483>:	jmpq   *%rax
   0x00007ffff5168aa5 <+485>:	nopl   (%rax)
   0x00007ffff5168aa8 <+488>:	add    $0x58,%rsp
   0x00007ffff5168aac <+492>:	pop    %rbx
   0x00007ffff5168aad <+493>:	pop    %rbp
   0x00007ffff5168aae <+494>:	pop    %r12
   0x00007ffff5168ab0 <+496>:	pop    %r13
   0x00007ffff5168ab2 <+498>:	pop    %r14
   0x00007ffff5168ab4 <+500>:	pop    %r15
   0x00007ffff5168ab6 <+502>:	retq   
End of assembler dump.
(gdb) quit
```

**Comment by Krozark** [comment]
same thing with vosk==0.3.15 (in the same venv after updating)

```
python ./test_empty.py 
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:192) Decoding params beam=13 max-active=7000 lattice-beam=6
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:195) Silence phones 1:2:3:4:5:6:7:8:9:10
LOG (VoskAPI:RemoveOrphanNodes():nnet-nnet.cc:948) Removed 1 orphan nodes.
LOG (VoskAPI:RemoveOrphanComponents():nnet-nnet.cc:847) Removing 2 orphan components.
LOG (VoskAPI:Collapse():nnet-utils.cc:1472) Added 1 components, removed 2
LOG (VoskAPI:CompileLooped():nnet-compile-looped.cc:345) Spent 0.00944996 seconds in looped compilation.
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:219) Loading i-vector extractor from /media/HDD1/DEV/Projet/git/Ambience-voice-assistent/data/lang/fr/model/ivector/final.ie
LOG (VoskAPI:ComputeDerivedVars():ivector-extractor.cc:183) Computing derived variables for iVector extractor
LOG (VoskAPI:ComputeDerivedVars():ivector-extractor.cc:204) Done.
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:237) Loading HCLG from /media/HDD1/DEV/Projet/git/Ambience-voice-assistent/data/lang/fr/model/graph/HCLG.fst
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:256) Loading words from /media/HDD1/DEV/Projet/git/Ambience-voice-assistent/data/lang/fr/model/graph/words.txt
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:264) Loading winfo /media/HDD1/DEV/Projet/git/Ambience-voice-assistent/data/lang/fr/model/graph/phones/word_boundary.int
LOG (VoskAPI:ReadDataFiles():vosk/model.cc:272) Loading CARPA model from /media/HDD1/DEV/Projet/git/Ambience-voice-assistent/data/lang/fr/model/rescore/G.carpa
{'text': ''}
```

**Comment by nshmyrev** [comment]
Reported openblas issue, most likely we'll release another version soon.

**Comment by nshmyrev** [comment]
Should be fixed now in new release 0.3.17, please check and reopen if needed
